Compile failing - chaingen.cpp:569:140: error: no matching function for call to ‘service_nodes

While trying to compile on Debian buster (kernel 4.19.67-2+deb10u2) I end up with this error:

[ 95%] Built target daemon
/home/crypto/loki/tests/core_tests/chaingen.cpp: In member function ‘bool loki_chain_generator::create_block(loki_blockchain_entry&, uint8_t, const loki_blockchain_entry&, const cryptonote::account_base&, uint64_t, std::vector<long unsigned int>&, const std::vector<cryptonote::transaction>&, const service_nodes::block_winner&) const’:
/home/crypto/loki/tests/core_tests/chaingen.cpp:569:140: error: no matching function for call to ‘service_nodes::service_node_list::state_t::update_from_block(const loki_chain_generator_db&, cryptonote::network_type, const std::set<service_nodes::service_node_list::state_t>&, <brace-enclosed initializer list>, cryptonote::block&, std::vector<cryptonote::transaction>&, std::nullptr_t)’
   entry.service_node_state.update_from_block(db_, cryptonote::FAKECHAIN, state_history_, {} /*alt_states*/, entry.block, entry.txs, nullptr);
                                                                                                                                            ^
In file included from /home/crypto/loki/src/rpc/core_rpc_server_commands_defs.h:45,
                 from /home/crypto/loki/src/cryptonote_core/tx_pool.h:48,
                 from /home/crypto/loki/src/cryptonote_core/cryptonote_core.h:42,
                 from /home/crypto/loki/tests/core_tests/chaingen.h:53,
                 from /home/crypto/loki/tests/core_tests/chaingen.cpp:49:
/home/crypto/loki/src/cryptonote_core/service_node_list.h:424:12: note: candidate: ‘void service_nodes::service_node_list::state_t::update_from_block(const cryptonote::BlockchainDB&, cryptonote::network_type, const std::set<service_nodes::service_node_list::state_t>&, const std::set<service_nodes::service_node_list::state_t>&, const std::unordered_map<crypto::hash, service_nodes::service_node_list::state_t>&, const cryptonote::block&, const std::vector<cryptonote::transaction>&, const keys_ptr&)’
       void update_from_block(
            ^~~~~~~~~~~~~~~~~
/home/crypto/loki/src/cryptonote_core/service_node_list.h:424:12: note:   candidate expects 8 arguments, 7 provided
make[3]: *** [tests/core_tests/CMakeFiles/core_tests.dir/build.make:115: tests/core_tests/CMakeFiles/core_tests.dir/chaingen.cpp.o] Error 1
make[3]: Leaving directory '/home/crypto/loki/build/Linux/master/release'
make[2]: *** [CMakeFiles/Makefile2:4376: tests/core_tests/CMakeFiles/core_tests.dir/all] Error 2
make[2]: Leaving directory '/home/crypto/loki/build/Linux/master/release'
make[1]: *** [Makefile:141: all] Error 2
make[1]: Leaving directory '/home/crypto/loki/build/Linux/master/release'
make: *** [Makefile:104: release-all] Error 2

--- a/tests/core_tests/chaingen.cpp
+++ b/tests/core_tests/chaingen.cpp
@@ -566,7 +566,7 @@ bool loki_chain_generator::create_block(loki_blockchain_entry &entry,
   fill_nonce(blk, TEST_DEFAULT_DIFFICULTY, height);
   entry.txs = tx_list;
   entry.service_node_state = prev.service_node_state;
-  entry.service_node_state.update_from_block(db_, cryptonote::FAKECHAIN, state_history_, {} /*alt_states*/, entry.block, entry.txs, nullptr);
+  entry.service_node_state.update_from_block(db_, cryptonote::FAKECHAIN, state_history_, {} /*state_archive*/, {} /*alt_states*/, entry.block, entry.txs, nullptr);
 
   uint64_t block_reward, block_reward_unpenalized;
   cryptonote::get_base_block_reward(epee::misc_utils::median(block_weights), entry.block_weight, prev.already_generated_coins, block_reward, block_reward_unpenalized, hf_version, height);

